### DISCUSSION TITLE
Support for modularity in Livewire.

### DISCUSSION THREAD

**Discussion by Modelizer** [livewire/livewire#3873] [1. Help]
Livewire is a great tool for projects where all the logic resides in app folder. But what if all the business logic is segregated to their own folder which consist of controllers, models, factories, tests, etc. For bigger projects organizing code plays a crucial factor and following the Domain-Driven Design principles provide us that leverage.

I would like to understand/know is there any possibility of allowing Livewire to integrate well with modularity? I know this will add complexity for a beginner but somehow there should be a way to keep Livewire flexible when the need arises.

Thanks!

**Comment by ariaieboy** [comment]
did you check this url :
[Registering Custom Components](https://laravel-livewire.com/docs/2.x/package-dev)
 you can register custom components using `Livewire::component` in the boot method of your service provider.
